lib/cb/utils/validator.rb

```diff
@@ -31,10 +31,12 @@ def raise_response_code_errors(response)
       end
 
       def fail_with_error_details(response, error_type)
-        error = error_type.new
+        processed_response = process_response_body(response)
+        error_message = processed_response['errors'] ? processed_response['errors'][0] : nil
```

Comment:
I don't know your opinion about it, but I've been tending towards using `.first` instead of an index lookup when I just want the first item. Nitpick/opinion, not something necessarily worth changing.